l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,449 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+import {Ownable2Step} from "@openzeppelin/contracts/access/Ownable2Step.sol";
+import {EfficientCall} from "@matterlabs/zksync-contracts/l2/system-contracts/libraries/EfficientCall.sol";
+import {IConsensusRegistry} from "./interfaces/IConsensusRegistry.sol";
+
+/// @author Matter Labs
+/// @custom:security-contact security@matterlabs.dev
+/// @title ConsensusRegistry
+/// @dev Manages consensus nodes and committees for the L2 consensus protocol,
+/// owned by Matter Labs Multisig. Nodes act as both validators and attesters,
+/// each playing a distinct role in the consensus process. This contract facilitates
+/// the rotation of validator and attester committees, which represent a subset of nodes
+/// expected to actively participate in the consensus process during a specific time window.
+contract ConsensusRegistry is IConsensusRegistry, Ownable2Step {
+    /// @dev An array to keep track of node owners.
+    address[] public nodeOwners;
+    /// @dev A mapping of node owner indices (index within the array of node owners) => node owners.
+    mapping(uint32 => address) public nodeOwnerIndices;
```

Comment:
Well, why you need this mapping at all? `nodeOwners` array stores the same information by index. `index -> address`.

Reply:
My bad 🤦
I overlook that the array value is indeed the mapping key.

Fixed here: https://github.com/matter-labs/era-contracts/pull/555/commits/8fd2100be48f19a04cc3dea414e0f77b40270db7

Reply:
My bad 🤦
Late night coding made me somehow think it will require to iterate over it.

Fixed here: https://github.com/matter-labs/era-contracts/pull/555/commits/8fd2100be48f19a04cc3dea414e0f77b40270db7